src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -127,7 +127,25 @@ public async Task<bool> RestoreAsync(string entryProjectFilePath, IDictionary<st
                     restorePC: IsOptionTrue(nameof(RestoreTaskEx.RestorePackagesConfig), options),
                     cleanupAssetsForUnsupportedProjects: IsOptionTrue(nameof(RestoreTaskEx.CleanupAssetsForUnsupportedProjects), options),
                     log: MSBuildLogger,
-                    cancellationToken: CancellationToken.None);
+                cancellationToken: CancellationToken.None);
+
+                // If the MSBuildBinaryLoggerEnabled environment variable is set, return the paths to all of the projects used during restore so that the assets file and other NuGet files
+                // can be embedded in the binlog
+                if (string.Equals(Environment.GetEnvironmentVariable("MSBUILDBINARYLOGGERENABLED"), bool.TrueString, StringComparison.OrdinalIgnoreCase))
+                {
+                    IReadOnlyList<PackageSpec> projects = dependencyGraphSpec.Projects;
```

Comment:
This covers all projects, which may included projects that don't contain nuget targets + packages.config projects etc. 

We can probably handle that better.